src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -262,7 +262,7 @@ final class HashMap[K, +V] private[immutable] (private[immutable] val rootNode:
           val (mergedK, mergedV) = mergef(payload, thatPayload)
           val mergedOriginalHash = mergedK.##
           val mergedImprovedHash = improve(mergedOriginalHash)
-          new HashMap(that.rootNode.updated(mergedK, mergedV, mergedOriginalHash, mergedImprovedHash, 0, replaceValue = true))
+          new HashMap(that.rootNode.removed(thatPayload._1, originalHash, improved, 0).updated(mergedK, mergedV, mergedOriginalHash, mergedImprovedHash, 0, replaceValue = true))
```

Comment:
does `removed` return `this` if neither `originalHash` nor `improved` match anything? if so, I think this is good. if not, I think it would be good to add a pre-check for that.

Reply:
(I'm not in a position to check right now)

Reply:
(yes it does turn this in that case)

Reply:
For this specific case, could we add a test that checks for reference equality?